samples/net/openthread/ncp/src/border-router.c

```diff
@@ -0,0 +1,23 @@
+/* border-router.c - OpenThread NCP Border Router */
```

Comment:
Could we call the file `main.c`? I've already expressed my concerns regarding calling the sample a border router.

Also, I believe we need some documentation for the sample. At minimum, we should link to the official OpenThread documentation (https://openthread.io/guides/border-router) and emphasize that this sample is the NCP part of the whole solution.

Reply:
documentation has been added.

border-router.c has been renamed to main.c.